File: InsuranceBackend/InsuranceProject/DTOs/PaymentDto.cs
using InsuranceProject.Types;
using System.ComponentModel.DataAnnotations;

namespace InsuranceProject.DTOs
{
    public class PaymentDto
    {
        public Guid PaymentId { get; set; }
        //public PaymentType? PaymentType { get; set; }
        public double Amount { get; set; }
        //public Guid? PremiumId { get; set; }
        public DateTime PaymentDate { get; set; }

        public int? indexId { get; set; }
        public double? Tax { get; set; }
        public double? TotalPayment { get; set; }
        
        public Guid? InsuranceSchemeId { get; set; }

        public int? PolicyNumber { get; set; }

        public Guid? AgentId { get; set; }
        public Guid PolicyId { get; set; }
        public string? Status { get; set; } = "Unpaid";
        public int? PoliciesCount { get; set; }
    }
}

File: InsuranceFrontEnd/InsuranceFrontEnd-main/src/app/models/PaymentDTO.ts
export class PaymentDTO {
    paymentType: number = 0
    paymentDate: Date = new Date()
    amount: number = 0
    tax: number = 0
    totalPayment: number = 0
    status: Boolean = true
    cardHolderName: string = ''
    cvvNo: string = ''
    cardNumber: string = ''
    expiryDate: Date = new Date()
    policyNumber!: number
}